datashuttle/utils/getters.py

```diff
@@ -339,3 +340,86 @@ def get_all_sub_and_ses_names(
             )
 
     return {"sub": all_sub_folder_names, "ses": all_ses_folder_names}
+
+
+def get_all_sub_and_ses(
+    self,
+    top_level_folder: str,
+    local_only: bool = False,
+) -> List[Dict[str, Union[str, List[Path]]]]:
+    """
+    Get all subjects and their session paths in the project.
+
+    Returns
+    -------
+    List[Dict]
+        Each dictionary contains:
+        - "subject": subject name (e.g., "sub-001")
+        - "session_paths": list of full Path objects to sessions.
+    """
+    project_info = []
+    sub_ses_names = get_all_sub_and_ses_names(
+        self.cfg, cast(TopLevelFolder, top_level_folder), local_only
+    )
+
+    for sub in sub_ses_names["sub"]:
+        session_paths = []
+
+        if local_only:
```

Comment:
This should check the dict within `sub_ses_names["sub"]` which I believe contains information on central vs local. For central, this would be `get_central_path()`.